src/pages/api/v3/locations/[id].ts

```diff
@@ -0,0 +1,120 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+
+import * as Location from "lib/database/location";
+import { HTTP_STATUS as STATUS } from "lib/httpStatusCodes";
+import { ZodError } from "zod";
+import { zIdFromString, zUpdateLocation } from 'lib/types/ZodSchemas';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  /**
+   * GET /api/v3/locations/[id]
+   **/
+  if (req.method === "GET") {
+    try {
+      /* parse parameters */
+      const locationId = zIdFromString.parse(req.query.id);
+
+      const location = await Location.findById({ id: locationId });
+
+      if (!location) {
+        const message = `Location with id '${locationId}' does not exist`;
+        console.log(`${req.method} /api/v3/locations/[id]:: ${message}`);
+
+        res.status(STATUS.NOT_FOUND)
+          .json({ message });
+        return;
+      }
+
+      res.status(STATUS.OK)
+        .json(location);
+    }
+
+    catch (e) {
```

Comment:
FIXME [ALL_API]: Perhaps build a custom logger that outputs the same format every time, currently we output different errors, e.g.:
```
400: { formErrors: [], fieldErrors: {} }
404: { message: "Resource not found" }
500: { error: "Internal server error" }

I will add this as an Issue on the repo for the future

Reply:
@juliusmarminge See my comment on the issue.